eclipse/src/saros/context/SarosEclipseContextFactory.java

```diff
@@ -77,18 +70,6 @@
       Component.create(ConnectingFailureHandler.class),
       Component.create(IncomingFileTransferHandler.class),
 
-      // Cache support
```

Comment:
No, the goal was to move it into the session context. Keeping it in the application context would not resolve any of the issues this is trying to resolve (cyclic dependencies and the filesystem listener only working during sessions due to using reference point based resources, potentially leading to outdated checksums that are seen as valid by the cache). To keep it in the application context, a different approach for the filesystem listener would be required (like making it generic; see https://github.com/saros-project/saros/commit/0162fb579738a07b853391b5cfadbe657c265dc3).

Reply:
So the current implementation would then help at least IntelliJ and the server as the checksum cache ould be used when you implement a file listener ?

But I still see no problem to refactor the logic in such a way that you can decide if you want to only track shared files (by just implementing a file listener) or use a complete different implementation.

Guess this can be decided after your whole work is merged.